nibabel/arrayproxy.py

```diff
@@ -336,36 +336,80 @@ def _get_fileobj(self):
                     self.file_like, keep_open=False) as opener:
                 yield opener
 
-    def get_unscaled(self):
-        """ Read of data from file
-
-        This is an optional part of the proxy API
-        """
-        with self._get_fileobj() as fileobj, self._lock:
-            raw_data = array_from_file(self._shape,
+    def _get_unscaled(self, slicer):
+        if canonical_slicers(slicer, self._shape, False) == \
+                canonical_slicers((), self._shape, False):
```

Comment:
This is a heavy check for whether we can just grab the whole file, which ensures we get an `mmap` if possible.

Might break some people using `dataobj[:]` to always retrieve an in-memory copy.